instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -40,6 +40,24 @@ def request(req, body = nil, &block)
 
             private
 
+            def connect
+              if proxy?
```

Comment:
Repeating this check to retrieve the hostname and port https://github.com/ruby/ruby/blob/master/lib/net/http.rb#L975-L981